katcp/testutils.py

```diff
@@ -699,10 +701,13 @@ def test_help(self, expected_requests, full_descriptions=False,
 
         if descriptions:
             if not full_descriptions:
-                got_requests = [(name, desc.split('\n')[0])
+                got_requests = [(name, desc.split(b'\n')[0])
                                 for (name, desc) in got_requests]
-                expected_requests = [(name, desc.split('\n')[0])
-                                     for (name, desc) in expected_requests]
+                expected_requests = [
+                    (native_str_to_bytes(name),
+                     native_str_to_bytes(desc).split(b"\n")[0])
+                    for (name, desc) in expected_requests
+                ]
         else:
             got_requests = [name for (name, desc) in got_requests]
```

Comment:
I think we also need to convert the expected requests if we're not doing descriptions.
```suggestion
            got_requests = [name for (name, desc) in got_requests]
            expected_requests = [native_str_to_bytes(name) for name in expected_requests]

```
Suggestion untested, so please try it out.